starter_code/src/movies.js

```diff
@@ -4,10 +4,51 @@
 
 // Get the average of all rates with 2 decimals 
 
+function ratesAverage(myArray) {
+  var totalAverage = 
+  myArray.reduce(function (acc, oneNumber) {
+    return (acc + oneNumber.rate) ;
+  }, 0);
+  var average = totalAverage / myArray.length ; 
+  return average;
+}
+
+// Erreur Paul - to not consider
+// movies.reduce(function ratesAverage (acc, oneNumber) {
+//   return (acc + oneNumber.rate) /   oneNumber.length;
+// }, 0);
+
 
 // Get the average of Drama Movies
 
 
+var dramaOk =
+movies.filter(function(film) {
+  return film.genre.indexOf('Drama') != -1;
+})
+
+var dramaRatesAverage = dramaOk.reduce(function(sum, note){
+  return sum + note.rate;
+}, 0);
+
+console.log(dramaRatesAverage);
```

Comment:
The logic is good! You have to divide your dramaRatesAverage by dramaOk.length to have the average rate, end wrap all of this code inside a function : 
```
function dramaMoviesRate(movies) {
  var dramaOk = movies.filter(function(film) {
     return film.genre.indexOf('Drama') != -1;
  })

  var dramaRatesTotal = dramaOk.reduce(function(sum, note){
     return sum + note.rate;
  }, 0);

  var dramaRatesAverage = dramaRatesTotal/dramaOK.length
 return dramaRatesAverage
}
```